### PR TITLE
feat: add includeEnvironment parameter #870

### PR DESCRIPTION
**Pull request by massimeddu-sj**
This PR introduce a new parameter `includeEnvironment` that by default is true.

### CODE REVIEW

**Comment by monperrus** [comment]
Thanks for the PR, we'll look at it.

**Comment by monperrus** [comment]
CI is failing?

**Comment by massimeddu-sj** [comment]
> CI is failing?

Apparently some tests are failing, but currently I'm not able to reproduce these failures on my local environment. I'll try to understand why they are failing and I'll try to fix them.

**Comment by massimeddu-sj** [comment]
> CI is failing?

I've fixed the failing tests and the formatting issues. Now it should be good.

**Comment by massimeddu-sj** [comment]
I've rebased the PR to solve merge conflicts.

**Comment by LogFlames** [comment]
Thank you for the PR! 
Please take a look at my one comment, the rest of the code looks good. 
I will also look into the ˋcheck-lockfileˋ action.

**Comment by massimeddu-sj** [comment]
> Thank you for the PR! Please take a look at my one comment, the rest of the code looks good. I will also look into the ˋcheck-lockfileˋ action.

Sorry, I cannot find your comment. Can you please link it or re-post here?

**Comment by LogFlames** [comment]
Of course, [here it is](https://github.com/chains-project/maven-lockfile/pull/871#discussion_r1746778701)

**Comment by LogFlames** [comment]
Yes your right, I hadn’t published the review. Now it should be visible!

File: maven_plugin/src/test/java/it/IntegrationTestsIT.java [hunk 1]
```diff
@@ -1,6 +1,7 @@
 package it;
 
 import static com.soebes.itf.extension.assertj.MavenITAssertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThat;
```
[the remark this answers is not in the document]
Comment (massimeddu-sj) [reply maven_plugin/src/test/java/it/IntegrationTestsIT.java]: Definitly not needed. Possibly it was imported automatically by my IDE. I've removed it.

[the remark this answers is not in the document]
Comment (LogFlames) [reply maven_plugin/src/test/java/it/IntegrationTestsIT.java]: Perfect!

**Pull request merged**